dialogue-core/src/main/java/com/palantir/dialogue/core/QueueOverrideChannel.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+
+final class QueueOverrideChannel implements Channel {
+
+    private final Channel defaultDelegate;
+
+    QueueOverrideChannel(Channel defaultDelegate) {
+        this.defaultDelegate = defaultDelegate;
+    }
+
+    @Override
+    public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        Channel override = request.attachments().getOrDefault(QueueAttachments.QUEUE_OVERRIDE, null);
+        if (override != null) {
+            return override.execute(endpoint, request);
+        } else {
+            return defaultDelegate.execute(endpoint, request);
+        }
```

Comment:
No change requested, defer to your judgement:

The `@Nullable` return value of `getOrDefault` is a little odd here, isn't it. I'd want to write this using the default, but that angers `NullAway`:
```suggestion
        Channel channel = request.attachments().getOrDefault(QueueAttachments.QUEUE_OVERRIDE, defaultDelegate);
        return channel.execute(endpoint, request);
```

Reply:
Yeah, sadly there's no way to tell the null check that the result is null IFF the argument is null? I copied the design for Attachments#put from Iterables#getFirst in terms of the annotations (i.e. both arg and return type are nullable)

Reply:
A Preconditions.checkNotNull around the call would do it, but it's pretty silly....